### DISCUSSION TITLE
Data copy on non-prod branches

### DISCUSSION THREAD

**Discussion by saihaj** [planetscale/discussion#112] [Feature Requests] [answered]
When a new branch is created there is no data from prod. Is there a way I can copy data over or just fill it partially?

**Comment by jgreet** [comment, marked as answer]
Currently new database branches contain only the schema from the parent branch. We plan on releasing a data branching feature in the future.

**Reply by saihaj** [reply]
Is there a way for me to download database backups so I can use those to fill data in new branch? I guess I could also connect to prod and do a `mysqldump` ?

**Reply by jgreet** [reply]
You could use mysqldump, but you could also restore a backup to a new branch. It'll be slower than regular branching, but it will copy both the schema and the data from the backup to the new branch.

You can restore a backup from the web app by clicking through to the list of backups for your main branch, or on the command line with pscale:

```shell
$ pscale backup restore <your-db-name> <your-new-branch-name> <12-character-backup-id>
```

**Reply by saihaj** [reply]
Thanks! that will solve my use case for now
